extensions/Lily/MoreEvents.js

```diff
@@ -205,6 +212,33 @@
         runtime.shouldExecuteStopClicked = false;
         originalGreenFlag.call(this);
       };
+
+      let pauseHandler = () => {
+        runtime.allScriptsByOpcodeDo(
+          "lmsMoreEvents_whenProjectPaused",
+          (script, target) => {
+            const topBlockId = script.blockId;
+            for (let j = 0; j < runtime.threads.length; j++) {
+              if (
+                runtime.threads[j].topBlock === topBlockId &&
+                runtime.threads[j].status !== 4
+              ) {
+                return;
+              }
+            }
+            runtime._pushThread(topBlockId, target);
+          }
+        );
+        runtime.removeListener("BEFORE_EXECUTE", pauseHandler);
+      };
+
+      runtime.on("RUNTIME_PAUSED", () => {
+        runtime.on("BEFORE_EXECUTE", pauseHandler);
+      });
+
+      runtime.on("RUNTIME_UNPAUSED", () => {
+        runtime.startHats("lmsMoreEvents_whenProjectUnPaused");

```

Comment:
It's possible for someone to pause then unpause in the same frame (eg.  if they lower the framerate).

It seems like what will happen here is that because RUNTIME_PAUSED handler until BEFORE_EXECUTE to start threads while the RUNTIME_UNPAUSED handler does startHats immediately, the when unpaused scripts will run before the when paused scripts, which is backwards. I guess they should both be in BEFORE_EXECUTE to ensure consistent ordering